applications/nrf_desktop/doc/usb_state.rst

```diff
@@ -6,3 +6,66 @@ USB state module
++--------------------+----------------------------+             +-----------------------------------+------------------+
++ ``hid_state``      + ``hid_report_event``       +             + ``hid_report_sent``               + ``hid_state``    +
++--------------------+----------------------------+             +-----------------------------------+                  +
++                    + ``config_event``           +             + ``hid_report_subscription_event`` +                  +
++--------------------+----------------------------+             +-----------------------------------+------------------+
++                    + ``config_fetch_event``     +             + ``usb_state_event``               +                  +
++--------------------+----------------------------+             +-----------------------------------+------------------+
++                    + ``config_forwarder_event`` +             +                                   +                  +
++--------------------+----------------------------+-------------+-----------------------------------+------------------+
+
+
+Configuration
+*************
+
+The ``usb_state`` is enabled by selecting ``CONFIG_DESKTOP_USB_ENABLE``.
+It depends on :option:`CONFIG_USB_DEVICE_HID`.
+
+When enabling the USB support for the device, set the following generic options:
+
+ * :option:`CONFIG_USB_DEVICE_MANUFACTURER` - the device manufacturer's name
+ * :option:`CONFIG_USB_DEVICE_PRODUCT` - the device's product name
+ * :option:`CONFIG_USB_DEVICE_VID` - the device's vendor ID number
+ * :option:`CONFIG_USB_DEVICE_PID` - the device`s product ID number
+
+For low latency devices, make sure that the device requests a polling rate of 1 ms by setting :option:`CONFIG_USB_HID_POLL_INTERVAL_MS` to 1.
+
+If the device is meant to support the boot protocol, select :option:`CONFIG_USB_HID_BOOT_PROTOCOL` and make sure :option:`CONFIG_USB_HID_PROTOCOL_CODE` chooses between a mouse or a keyboard code.
+
+Implementation details
+**********************
+
+The ``usb_state`` module registers the HID-class USB device and initializes the USB stack.
+
+The necessary callbacks are connected to ensure that the state of the USB connection is tracked.
+From the application's viewpoint, the USB can be in the following states, which are broadcast by the ``usb_state`` module with a ``usb_state_event``:
+
+ * USB_STATE_DISCONNECTED - USB cable is not connected.
+ * USB_STATE_POWERED - The device is powered from the USB but is not configured for the communication.
+ * USB_STATE_ACTIVE - The device is ready to exchange data with the host.
+ * USB_STATE_SUSPENDED - The host has requested the device to stop operating.
+
+When the device is connected to the host and configured for the communication, the module will broadcast ``USB_STATE_ACTIVE`` state with ``usb_state_event``.
+The module will also subscribe to all HID reports available in the application.
```

Comment:
Should we say something about the boot protocol here?
In boot mode we subscribe only for the boot report.

Reply:
I would no go so much into details here.

Reply:
I will add "available for selected protocol" - this should be fine.

Reply:
Done